### ISSUE TITLE
Gutenberg: Inline Image block's media modal fires a ton of `onOpen`

### ISSUE THREAD

**Issue by Copons** [Automattic/wp-calypso#27818]
Follow up to #27687

When opening the `Inline Image` block's media modal, the `onOpen` function is called _a lot_ of times.
The `setState` contained inside is called as many times, to the point that it crashes the modal.

I couldn't figure out why (in Core it is called only once, but then again, the logic is very different), but for now I've worked around this issue by [only calling `setState` if needed](https://github.com/Automattic/wp-calypso/pull/27687/files#diff-0db2c9807d11b1264b1631f5e42f5dd9R44).

**Comment by vindl** [comment]
This was an issue specific to the Media Modal integration in Gutenlypso. After switching to Gutenframe it's no longer relevant.
